imports/client/components/HuntListPage.tsx

```diff
@@ -33,6 +34,109 @@ const splitLists = function (lists: string): string[] {
+  ready: boolean;
+  discordChannels: DiscordChannelType[];
+}
+
+class DiscordChannelForm extends React.Component<DiscordChannelFormProps> {
+  onPuzzleHooksDiscordChannelChanged: FormControlProps['onChange'] = (e) => {
+    if (e.currentTarget.value === 'empty') {
+      this.props.onChange(undefined);
+    } else {
+      const match = this.props.discordChannels.find((chan) => { return chan._id === e.currentTarget.value; });
+      if (match) {
+        const next = {
+          id: e.currentTarget.value,
+          name: match.name,
+        };
+        this.props.onChange(next);
+      }
+    }
+  };
+
+  formOptions = (): SavedDiscordChannelType[] => {
+    // List of the options.  Be sure to include the saved option if it's (for
+    // some reason) not present in the channel list.
+    const noneOption = {
+      id: 'empty',
+      name: 'disabled',
+    } as SavedDiscordChannelType;
+
+    const propsListOptions = this.props.discordChannels.map((chan) => {
+      return {
+        id: chan._id,
+        name: chan.name,
+      };
+    });
+
+    if (this.props.puzzleHooksDiscordChannel) {
+      if (!propsListOptions.find((opt) => {
+        return opt.id === this.props.puzzleHooksDiscordChannel!.id;
+      })) {
+        return [noneOption, this.props.puzzleHooksDiscordChannel, ...propsListOptions];
+      }
+    }
+    return [noneOption, ...propsListOptions];
+  };
+
+  render() {
+    if (!this.props.ready) {
+      return <div>Loading discord channels...</div>;
+    } else {
+      return (
+        <FormGroup as={Row}>
+          <FormLabel column xs={3} htmlFor={`${this.props.idPrefix}puzzle-hooks-discord-channel`}>
+            Puzzle notifications Discord channel
+          </FormLabel>
+          <Col xs={9}>
+            <FormControl
+              id={`${this.props.idPrefix}puzzle-hooks-discord-channel`}
+              as="select"
+              type="text"
+              placeholder=""
+              value={this.props.puzzleHooksDiscordChannel && this.props.puzzleHooksDiscordChannel.id}
+              disabled={this.props.disableForm}
+              onChange={this.onPuzzleHooksDiscordChannelChanged}
+            >
+              {this.formOptions().map(({ id, name }) => {
+                return (
+                  <option key={id} value={id}>{name}</option>
+                );
+              })}
+            </FormControl>
+            <FormText>
+              If this field is specified, when a puzzle in this hunt is added or solved, a message will be sent to the specified channel.
+            </FormText>
+          </Col>
+        </FormGroup>
+      );
+    }
+  }
+}
+
+const DiscordChannelFormContainer = withTracker((_params: DiscordChannelFormParams) => {
+  const discordChannelsHandle = Meteor.subscribe('discord.channels');
+  const discordChannels = DiscordChannels.find(
+    // We want only text channels, since those are the only ones we can bridge chat messages to.
+    { type: 0 },
```

Comment:
OOC can we pull this constant from discord.js?

Reply:
I don't see it in the docs.  Looks like they have separate types and translate to strings in the public API: https://discord.js.org/#/docs/main/stable/class/Channel?scrollTo=type